test/distributed/_tensor/test_math_ops.py

```diff
@@ -393,6 +421,147 @@ def _replicate_fn(name, module, device_mesh):
+
+        subtest_fails = {}
+        valid_filter = lambda cfg: not (  # noqa: E731
+            cfg.ln_req_grad and not cfg.elementwise_affine
+        ) and any(cfg[2:])
+        subtest_cfgs = list(
+            filter(
+                valid_filter,
+                [SubTest(*cfg) for cfg in itertools.product(*(((False, True),) * 5))],
+            )
+        )
+
+        for subtest_cfg in subtest_cfgs:
+            try:
+                (
+                    multidim_norm,
+                    elementwise_affine,
+                    emb_req_grad,
+                    ln_req_grad,
+                    out_req_grad,
+                ) = subtest_cfg
+                normalized_shape = (
+                    (seq_len, embedding_dim) if multidim_norm else (embedding_dim,)
+                )
+
+                # configure our local and parallelized models for this subtest
+                class LnTpBlock(torch.nn.Module):
+                    def __init__(self):
+                        super().__init__()
+                        self.preln_embeddings = torch.nn.Embedding(
+                            vocab_size, embedding_dim
+                        )
+                        self.layer_norm = torch.nn.LayerNorm(
+                            normalized_shape, elementwise_affine=elementwise_affine
+                        )
+                        self.postln_linear = torch.nn.Linear(
+                            embedding_dim, embedding_dim
+                        )
+
+                    def forward(self, tokens):
+                        h = self.preln_embeddings(tokens)
+                        h = self.layer_norm(h)
+                        output = self.postln_linear(h)
+                        return output
+
+                parallel_plan = {
+                    "preln_embeddings": RowwiseParallel(
+                        input_layouts=Replicate(), output_layouts=Shard(1)
+                    ),
+                    "layer_norm": SequenceParallel(),
+                    "postln_linear": ColwiseParallel(
+                        input_layouts=Shard(1),
+                        output_layouts=Replicate(),
+                    ),
+                }
+
+                model = LnTpBlock().to(device=self.device_type)
+                model_local = copy.deepcopy(model).to(device=self.device_type)
+                model_dist = parallelize_module(model, device_mesh, parallel_plan)
+                req_grad_map = {
+                    "preln_embeddings": emb_req_grad,
+                    "postln_linear": out_req_grad,
+                    "layer_norm": ln_req_grad,
+                }
+
+                # apply the relevant `requires_grad` mask for this subtest to both models
+                for target_model in [model_local, model_dist]:
+                    for n, p in target_model.named_parameters():
+                        if not req_grad_map.get(n.rpartition(".")[0], False):
+                            p.requires_grad_(False)
```

Comment:
can we add a check on the `requires_grad` field after this to make sure that the setting is correct?

Reply:
Done!